### PR TITLE
Update readme

### PR DESCRIPTION
**Pull request by arikmaor**
Updates installation instructions based on #31 and running first local installation with @tip3x

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,18 +1,58 @@
...
+gsutil cp gs://tensorleap-infra-nonprod/ops/default.tfstate - | jq -r '.resources[] | select(.name == "ops_key" and .type == "github_actions_organization_secret") | .instances[].attributes.plaintext_value' > ./json-key-file.json
+kubectl --kubeconfig=<(microk8s config) create secret docker-registry gcr-access-token \
+  --docker-server=gcr.io \
+  --docker-username=_json_key \
+  --docker-password="$(cat ./json-key-file.json)" \
+  --docker-email=someone@tensorleap.ai
+rm ./json-key-file.json
+
+# deploy tensorleap to local cluster
+kubectl apply -f ./dist
```
Comment (roee-shapira-tensorleap) [inline README.md:47]: Maybe you could use `kapp` here?

File: README.md [hunk 2]
```diff
@@ -1,18 +1,58 @@
...
+
+# Build manifests
+npm install
+npm run build
+
+## ========= MAKE SURE YOU'RE IN MICROK8S CONTEXT ===========
+kubectx -c # => should print microk8s
+
+# get ops service account secret and deploy it to microk8s cluster
+gsutil cp gs://tensorleap-infra-nonprod/ops/default.tfstate - | jq -r '.resources[] | select(.name == "ops_key" and .type == "github_actions_organization_secret") | .instances[].attributes.plaintext_value' > ./json-key-file.json
```
Comment (roee-shapira-tensorleap) [inline README.md:38]: Add the `brew install` command for this?

File: README.md [hunk 3]
```diff
@@ -1,18 +1,58 @@
...
 
+Tensorleap is served in http://tensorleap.local \
 Create a user using signup requests in http://tensorleap.local/api/v2/swagger
 
 ### On Ubuntu
 
-```
+```bash
 sudo apt update
 sudo apt upgrade
```
Comment (roee-shapira-tensorleap) [inline README.md:57]: I think that it would be best to use only `apt` or only `apt-get` - not both

File: README.md [hunk 4]
```diff
@@ -1,18 +1,58 @@
...
 brew install ubuntu/microk8s/microk8s
 microk8s install
+microk8s status --wait-ready
 microk8s enable ingress dns storage rbac
+
+# Update /etc/hosts
+echo "$(multipass list --format json | jq -r '.list[] | select(.name == "microk8s-vm") | .ipv4 | first')\ttensorleap.local" | sudo tee -a /etc/hosts
+
+# Add microk8s context to your local KubeConfig file
+KUBECONFIG=<(microk8s config):$HOME/.kube/config kubectl config view --raw > kubeConfig && mv kubeConfig $HOME/.kube/config
```
Comment (roee-shapira-tensorleap) [inline README.md:18]:
```suggestion
echo "$(KUBECONFIG=<(microk8s config):$HOME/.kube/config kubectl config view --raw)" | tee $HOME/.kube/config
```
Simpler? (but needs testing)

File: README.md [hunk 5]
```diff
@@ -1,18 +1,58 @@
...
+
+# install helm and add repositories
+brew install helm
+helm repo add elastic https://helm.elastic.co
+helm repo add bitnami https://charts.bitnami.com/bitnami
+helm repo add minio https://charts.min.io/
+
+# Build manifests
+npm install
+npm run build
```
Comment (roee-shapira-tensorleap) [inline README.md:32]: I this in the right dir?

File: README.md (hunk 2 above)
Comment (roee-shapira-tensorleap) [inline README.md:38]:
```suggestion
gsutil cp gs://tensorleap-infra-nonprod/ops/default.tfstate - | jq -r '.resources[] | select(.name == "ops_key" and .type == "github_actions_organization_secret") | .instances[].attributes.plaintext_value' > /tmp/json-key-file.json
```
?

(plus for the commands that use this file)

File: README.md [hunk 6]
```diff
@@ -79,7 +116,14 @@ sudo usermod -a -G microk8s $USER
...
 microk8s enable ingress dns storage rbac gpu
 microk8s enable host-access:ip=10.0.1.20
-# add a line `10.0.1.20<TAB>tensorleap.local` to `/etc/hosts`
+echo "10.0.1.20\ttensorleap.local" | sudo tee -a /etc/hosts
+
+# Get a service account key file with read permissions to storage and add to cluster
+kubectl --kubeconfig=<(microk8s config) create secret docker-registry gcr-access-token \
+  --docker-server=gcr.io \
+  --docker-username=_json_key \
+  --docker-password="$(cat ./json-key-file.json)" \
```
Comment (roee-shapira-tensorleap) [inline README.md:125]: I don't think that the `./json-key-file.json` is created in this section...

**Review by roee-shapira-tensorleap — COMMENTED** [review]
I didn't test this in my machine (I'm having some critical problems with `microk8s`) so I didn't approve it yet - but talk to me about this and I'll approve

File: README.md (hunk 4 above)
Comment (arikmaor) [reply README.md:18]: could break for some reason, tried that (but without `tee`)...
maybe `tee` works but we need to check (if you decided to check, back your original `KubeConfig`)

File: README.md (hunk 5 above)
Comment (arikmaor) [reply README.md:32]: In the main repo dir...

File: README.md (hunk 2 above)
Comment (arikmaor) [reply README.md:38]: I think people here have `gcloud` installed...

Comment (arikmaor) [reply README.md:38]: The next command use this file
The one after deletes it...

File: README.md (hunk 1 above)
Comment (arikmaor) [reply README.md:47]: that what I wanted to do, 10x for catching that!

File: README.md (hunk 3 above)
Comment (arikmaor) [reply README.md:57]: I agree, but this is what we did and worked
Next time I'm installing a laptop from scratch I'll try to change after it is tested

File: README.md (hunk 6 above)
Comment (arikmaor) [reply README.md:125]: That's why the comment

File: README.md (hunk 2 above)
Comment (roee-shapira-tensorleap) [reply README.md:38]: Forgot this was for mac...

Comment (roee-shapira-tensorleap) [reply README.md:38]: I saw that, but I just don't like to add/edit files in a git repo...

File: README.md (hunk 6 above)
Comment (roee-shapira-tensorleap) [reply README.md:125]: Ohhhh

**Review by tip3x — APPROVED** [review]
LGTM. Tested on my mac

File: README.md (hunk 2 above)
Comment (arikmaor) [reply README.md:38]: we can test and change in a separate PR

**Pull request merged**